### PR TITLE
Don't add -d by default. Parse options for stdout/stderr files.

### PR DESCRIPTION
**Pull request by asuonpaa**
Fixes https://github.com/google/amber/issues/706

### CODE REVIEW

File: samples/android_main.cc [hunk 1]
```diff
@@ -24,22 +24,44 @@ extern "C" JNIEXPORT JNICALL int Java_com_google_amber_Amber_androidMain(
...
-  for (auto s : argv_string)
-    argv.push_back(s.c_str());
+  std::string stdout_file = "/sdcard/amber_stdout.txt";
+  std::string stderr_file = "/sdcard/amber_stderr.txt";
+
+  for (size_t i = 0; i < argv_string.size(); i++) {
+    auto s = argv_string[i];
+
+    if (s == "--stdout") {
+      i++;
```
Comment (dj2) [inline samples/android_main.cc]: Need to validate i is < argv_string.size()

File: samples/android_main.cc [hunk 2]
```diff
@@ -24,22 +24,44 @@ extern "C" JNIEXPORT JNICALL int Java_com_google_amber_Amber_androidMain(
...
+      stdout_file = argv_string[i];
+      continue;
+    }
+
+    if (s == "--stderr") {
+      i++;
```
Comment (dj2) [inline samples/android_main.cc]: ditto

File: samples/android_main.cc [hunk 3]
```diff
@@ -23,23 +23,34 @@ extern int main(int argc, const char** argv);
...
   // Redirect std output to a file
-  freopen("/sdcard/amberlog.txt", "w", stdout);
-  freopen("/sdcard/amberlog.txt", "a", stderr);
+  freopen(stdout_file_cstr, "w", stdout);
+  freopen(stderr_file_cstr, "w", stderr);
+
+  env->ReleaseStringUTFChars(stdoutFile, stdout_file_cstr);
+  env->ReleaseStringUTFChars(stderrFile, stderr_file_cstr);
+
+  int arg_count = env->GetArrayLength(args);
```
Comment (paulthomson) [inline samples/android_main.cc]: Nit but could you use `jsize` here and below.

File: samples/android_main.cc [hunk 4]
```diff
@@ -23,23 +23,34 @@ extern int main(int argc, const char** argv);
...
+    argv_string.push_back(arg_cstr);
+    env->ReleaseStringUTFChars(js, arg_cstr);
+  }
 
-  // Parse argument string and add -d by default
-  std::stringstream ss(args);
-  std::vector<std::string> argv_string{std::istream_iterator<std::string>{ss},
-                                       std::istream_iterator<std::string>{}};
   std::vector<const char*> argv;
   argv.push_back("amber");
```
Comment (paulthomson) [inline samples/android_main.cc]: Nit but could you push "amber" into `argv_string` above (before the loop).

File: samples/android_main.cc [hunk 5]
```diff
@@ -23,23 +23,34 @@ extern int main(int argc, const char** argv);
...
-  argv.push_back("-d");
 
-  for (auto s : argv_string)
-    argv.push_back(s.c_str());
+  for (size_t i = 0; i < argv_string.size(); i++)
```
Comment (paulthomson) [inline samples/android_main.cc]: Nit but could you use a `for(const std::string& arg : argv_string)` loop instead?

Comment (asuonpaa) [reply samples/android_main.cc]: Sure

File: samples/android_main.cc (hunk 4 above)
Comment (asuonpaa) [reply samples/android_main.cc]: done

File: samples/android_main.cc (hunk 3 above)
Comment (asuonpaa) [reply samples/android_main.cc]: done

**Pull request merged**